src/overseer/store/jdbc.clj

```diff
@@ -166,9 +172,15 @@
                         (map job->jdbc-map)
                         (map #(assoc % :created_at now :updated_at now)))
           dep-rows (job-dep-jdbc-maps graph)]
-      (j/with-db-transaction [t-con db-spec]
-        (j/insert-multi! t-con :overseer_jobs job-rows)
-        (j/insert-multi! t-con :overseer_dependencies dep-rows))))
+      (try
+        (j/with-db-transaction [t-con db-spec]
+          (j/insert-multi! t-con :overseer_jobs job-rows)
+          (j/insert-multi! t-con :overseer_dependencies dep-rows))
+        graph
+        (catch Exception ex
```

Comment:
I know this is a little weird but I propose Rust-style Exception type hoisting.  This should really read like

```clojure
(try
  ...
  (catch PrimaryKeyException ex
    ...)
```
And not have this layer of the code exposed to that whackiness.  In other words that the generic exception is normalized by some other layer into a coherent set of standard exceptions for this (JDBC) context.

It may require some macroing:

```clojure
(with-db-exceptions ;; anything in this context has the proper exceptions available
  (try 
    ...
    (catch PrimaryKeyException ex
      ...)))
```